core-api/src/main/kotlin/com/jakemoore/datakache/core/connections/mongo/changestream/ChangeStreamEventProcessor.kt

```
@@ -0,0 +1,386 @@
+            val bsonValue = documentKey["_id"]
+            when {
+                bsonValue?.isObjectId == true -> bsonValue.asObjectId().value.toHexString()
+                bsonValue?.isString == true -> bsonValue.asString().value
+                bsonValue?.isInt32 == true -> bsonValue.asInt32().value
+                bsonValue?.isInt64 == true -> bsonValue.asInt64().value
+                else -> {
+                    context.logger.warn("Unsupported ID type in document key: ${bsonValue?.bsonType}")
+                    null
+                }
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)